zookeeper-server/src/main/java/org/apache/zookeeper/Op.java

```diff
@@ -146,10 +236,33 @@ public static Op create(String path, byte[] data, List<ACL> acl, CreateMode crea
      *                optional ttl or 0 (createMode must imply a TTL)
      */
     public static Op create(String path, byte[] data, List<ACL> acl, CreateMode createMode, long ttl) {
+        return create(path, data, acl, createMode, CreateFlags.DEFAULT, ttl);
+    }
+
+    /**
+     * Constructs a create operation.  Arguments are as for the ZooKeeper method of the same name
+     * but adding an optional ttl
+     * @see ZooKeeper#create(String, byte[], java.util.List, CreateMode)
+     *
+     * @param path
+     *                the path for the node
+     * @param data
+     *                the initial data for the node
+     * @param acl
+     *                the acl for the node
+     * @param createMode
+     *                specifying whether the node to be created is ephemeral
+     *                and/or sequential
+     * @param createFlags
+     *                the set of {@link CreateFlags} to apply
+     * @param ttl
+     *                optional ttl or 0 (createMode must imply a TTL)
+     */
+    public static Op create(String path, byte[] data, List<ACL> acl, CreateMode createMode, Set<CreateFlags> createFlags, long ttl) {
```

Comment:
adding a new factory method is not very awesome from my point of view.

can we take the opportunity to add a fluent Builder for "Op" ?
also adding "Deprecated" to the other factory methods

Reply:
Added a note below: https://github.com/apache/zookeeper/pull/1559#issuecomment-759411279

Reply:
I am leaning toward adding the Builder support without marking the methods as Deprecated (in order to not change tests, we have a few important big forks and changing lots of tests will be a pain for every one).

So adding the Builder and not adding this new overloaded method here.

I am not sure that adding the Builder is about adding much code. 

cc @hanm @lvfangmin

Reply:
> I am leaning toward adding the Builder support without marking the methods as Deprecated (in order to not change tests, we have a few important big forks and changing lots of tests will be a pain for every one).
> 
> So adding the Builder and not adding this new overloaded method here.

Right.  That's what I did in this other branch:

https://github.com/ztzg/zookeeper/commits/ZOOKEEPER-4026-multi-create2-fluent-builder

In particular, here is the builder for `Create` and `CreateTTL`:

https://github.com/ztzg/zookeeper/commit/d7bab23d8faf2f364a8bbe32a2c15170030e4d40#diff-d6e7807a04d07a070ee3da75ccd793bd973b5af0ee5597f0969a0a7721d0200fR376

> I am not sure that adding the Builder is about adding much code.

There are *six* direct subclasses of `Op`, though: https://javadoc.io/doc/org.apache.zookeeper/zookeeper/latest/org/apache/zookeeper/Op.html .  (The only indirect one is `CreateTTL`, and it is already handed.)

But looking closer, the others do not warrant introducing a `*Builder`, as they don't have the same variability in parameters.  Would you be okay with the asymmetry?  (I.e., only introducing a `CreateBuilder`, and leaving the rest as-is.)

Reply:
Okay—the version I just pushed implements the "fluent builder" pattern for `Create`, and only uses a simple `boolean` for the new feature, as pondered in https://github.com/apache/zookeeper/pull/1559#discussion_r556586446.

I find the result simpler and easier to read.  What do you think?